storagedriver/s3/s3.go

```diff
@@ -580,6 +580,16 @@ func (d *Driver) Delete(path string) error {
 	return nil
 }
 
+// URLFor returns a URL which may be used to retrieve the content stored at the given path.
+// May return an UnsupportedMethodErr in certain StorageDriver implementations.
+func (d *Driver) URLFor(path string) (string, error) {
+	if !storagedriver.PathRegexp.MatchString(path) {
+		return "", storagedriver.InvalidPathError{Path: path}
+	}
+
+	return d.Bucket.SignedURL(d.s3Path(path), time.Now().Add(24*time.Hour)), nil
```

Comment:
I added that to the discussion over at #49. Leave it out for now, and we can add it when a decision is made.